Core/MMKV_IO.cpp

```diff
@@ -1630,4 +1659,18 @@ size_t MMKV::filterExpiredKeys() {
     return count;
 }
 
+void MMKV::enableCompareBeforeSet() {
+    MMKVInfo("enableCompareBeforeSet for [%s]", m_mmapID.c_str());
+    SCOPED_LOCK(m_lock);
+    SCOPED_LOCK(m_exclusiveProcessLock);
+    m_enableCompareBeforeSet = true;
```

Comment:
When auto-key-expiration is enabled, is it OK to enable CompareBeforeSe? Furthermore, is OK to enable auto-key-expiration when compare-before-set is on?

Reply:
The relationship between compare-before-set & encryption also needs to be stated.

Reply:
Actually,  when encryption or expiration is on, compare-before-set should be invalid.
For encryption, compare operation must decrypt data which is time consuming
For expiration, compare is useless because in most cases expiration time changes every time.

So, what will I do to handle this two cases。Add some comment for this function?

Reply:
A runtime asserting would be better.

Reply:
Documentation should also make it clear, for sure.